### PR TITLE
Added some documentation about how rc works

### PR DESCRIPTION
**Pull request by mm-near**

### CODE REVIEW

File: core/store/src/columns.rs [hunk 1]
```diff
@@ -283,6 +282,21 @@ const OPTIONAL_GC_COLUMNS: [bool; DBCol::COUNT] = col_set(&[
...
+// Under the hood, we're using our custom merge operator (refcount_merge) to properly 'join' the refcounted cells.
+// WARNING: this means that the 'value' for a given key must never change.
+// Example:
+//   update_refcount("foo", "bar", 1);
+//   // good - after this call, the RC will be equal to 3.
+//   update_refcount("foo", "bar", 2);
+//   // bad - the value is still 'bar'.
+//   update_refcount("foo", "baz", 1);
+//   // ok - the value will be removed now. (as rc == 0)
+//   update_refcount("foo", "", -3)
```
Comment (matklad) [inline core/store/src/columns.rs:298]: I'd maybe put it on the `pub fn is_rc`, as that's the public API.

File: core/store/src/columns.rs [hunk 2]
```diff
@@ -283,6 +282,21 @@ const OPTIONAL_GC_COLUMNS: [bool; DBCol::COUNT] = col_set(&[
     DBCol::ColStateParts,
 ]);
 
+// The columns below are 'reference-counted'. This means, that we're storing additional
+// 8 bytes at the end of the payload with the current RC value.
```
Comment (matklad) [inline core/store/src/columns.rs:286]: Could you add a bit on the semantics of negative/zero RC counts? I remeber being puzzled a lot by this function:

```rust
/// Returns
/// (Some(value), rc) if rc > 0
/// (None, rc) if rc <= 0
pub fn decode_value_with_rc(bytes: &[u8]) -> (Option<&[u8]>, i64) {
```

File: core/store/src/columns.rs [hunk 3]
```diff
@@ -283,6 +282,21 @@ const OPTIONAL_GC_COLUMNS: [bool; DBCol::COUNT] = col_set(&[
...
+// WARNING: this means that the 'value' for a given key must never change.
+// Example:
+//   update_refcount("foo", "bar", 1);
+//   // good - after this call, the RC will be equal to 3.
+//   update_refcount("foo", "bar", 2);
+//   // bad - the value is still 'bar'.
+//   update_refcount("foo", "baz", 1);
+//   // ok - the value will be removed now. (as rc == 0)
+//   update_refcount("foo", "", -3)
+
```
Comment (mina86) [inline core/store/src/columns.rs:299]: `///` to make it a docstring.

**Review by mina86 — APPROVED** [review]

**Review by mzhangmzz — APPROVED** [review]

File: core/store/src/columns.rs (hunk 3 above)
Comment (mm-near) [reply core/store/src/columns.rs:299]: done

**Pull request merged**